WooCommerce/Classes/ViewRelated/Orders/OrderListViewModel.swift

```diff
@@ -311,29 +311,31 @@ final class OrderListViewModel {
         )
     }
 
-    func feedbackBannerSurveySource() -> SurveyViewController.Source? {
+    func feedbackBannerSurveySource(onCompletion: (SurveyViewController.Source) -> Void ) {
         if isCODEnabled && isIPPSupportedCountry {
+            fetchIPPTransactions()
 
-            let hasResults = IPPOrdersResultsController.fetchedObjects.isEmpty ? false : true
+            let hasWCPayResults = WCPayOrdersResultsController.fetchedObjects.isEmpty ? false : true
+            let WCPAYResultsCount = WCPayOrdersResultsController.fetchedObjects.count
+            let hasOneOrMoreWCPayTransactions = (WCPAYResultsCount >= 1) ? true : false
 
             /// In order to filter WCPay transactions processed through IPP within the last 30 days,
             /// we check if these contain `receipt_url` in their metadata, unlike those processed through a website,
             /// which doesn't
             ///
-            let IPPTransactionsFound = recentIPPOrdersResultsController.fetchedObjects.filter({
+            let recentIPPWCPayTransactionsFound = recentWCPayIPPResultsController.fetchedObjects.filter({
                 $0.customFields.contains(where: {$0.key == Constants.receiptURLKey }) &&
                 $0.paymentMethodTitle == Constants.paymentMethodTitle})
-            let IPPresultsCount = IPPTransactionsFound.count
-
-            if !hasResults {
-                return .IPP_COD
-            } else if IPPresultsCount < Constants.numberOfTransactions {
-                return .IPP_firstTransaction
-            } else if IPPresultsCount >= Constants.numberOfTransactions {
-                return .IPP_powerUsers
+            let recentWCPayResultsCount = recentIPPWCPayTransactionsFound.count
+
+            if !hasWCPayResults {
```

Comment:
You can consider using guards

```
guard hasWCPayResults else {
        onCompletion(.inPersonPaymentsCashOnDelivery)
}
```

Reply:
I'd prefer to leave the if/else for the moment just because I find them more readable in this scenario, happy to change them if you believe improves the changes though!